src/sim/circuit.cc

```diff
@@ -0,0 +1,554 @@
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+#include <assert.h>
+#include <algorithm>
+#include <unordered_map>
+
+#include "circuit.h"
+#include "quartz/gate/gate_utils.h"
+#include "quartz/parser/qasm_parser.h"
+#include "quartz/tasograph/tasograph.h"
+
+
+
+namespace quartz {
+
+// check type
+int num_iterations_by_heuristics(CircuitSeq *seq, int num_local_qubits,
```

Comment:
Is this copied from `src/test/test_simulation.cpp`? We might need to put it in some other files later.

Reply:
Yes, we can put the ILP/heuristics part somewhere else.